src/server/status_server.rs

```diff
@@ -210,6 +211,104 @@ impl StatusServer {
         Box::new(ok(res))
     }
 
+    pub fn dump_rsprof(
+        seconds: u64,
+        frequency: i32,
+    ) -> Box<dyn Future<Item = rsperftools::Report, Error = rsperftools::Error> + Send> {
+        let guard = rsperftools::PROFILER
+            .write()
+            .start_with_guard(frequency)
+            .unwrap();
+        info!(
+            "start profiling {} seconds with frequency {} /s",
+            seconds, frequency
+        );
+
+        let timer = GLOBAL_TIMER_HANDLE.clone();
+        Box::new(
+            timer
+                .delay(std::time::Instant::now() + std::time::Duration::from_secs(seconds))
+                .then(
+                    move |_| -> Box<
+                        dyn Future<Item = rsperftools::Report, Error = rsperftools::Error> + Send,
+                    > {
+                        let _ = guard;
+                        Box::new(match rsperftools::PROFILER.read().report() {
+                            Ok(report) => ok(report),
+                            Err(e) => err(e),
+                        })
+                    },
+                ),
+        )
+    }
+
+    pub fn dump_rsperf_to_resp(
+        req: Request<Body>,
+    ) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+        let query = match req.uri().query() {
+            Some(query) => query,
+            None => {
+                let response = Response::builder()
+                    .status(StatusCode::BAD_REQUEST)
+                    .body(Body::empty())
+                    .unwrap();
+                return Box::new(ok(response));
+            }
+        };
+        let query_pairs: HashMap<_, _> = url::form_urlencoded::parse(query.as_bytes()).collect();
+        let seconds: u64 = match query_pairs.get("seconds") {
+            Some(val) => match val.parse() {
+                Ok(val) => val,
+                Err(_) => {
+                    let response = Response::builder()
+                        .status(StatusCode::BAD_REQUEST)
+                        .body(Body::empty())
+                        .unwrap();
+                    return Box::new(ok(response));
+                }
+            },
+            None => 10,
+        };
+
+        let frequency: i32 = match query_pairs.get("frequency") {
```

Comment:
We can supply a best default value for this parameter.

Reply:
From Go

```
func StartCPUProfile(w io.Writer) error {
	// The runtime routines allow a variable profiling rate,
	// but in practice operating systems cannot trigger signals
	// at more than about 500 Hz, and our processing of the
	// signal is not cheap (mostly getting the stack trace).
	// 100 Hz is a reasonable choice: it is frequent enough to
	// produce useful data, rare enough not to bog down the
	// system, and a nice round number to make it easy to
	// convert sample counts to seconds. Instead of requiring
	// each client to specify the frequency, we hard code it.
	const hz = 100
```
We know that the maximum HZ maybe 500, so here we must avoid too large frequency. 

Another thing is Go uses 100, but I still prefer 99 by default :-)